server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1318,9 +1322,61 @@ public static Metadata fromXContent(XContentParser parser) throws IOException {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        Builder.toXContent(this, builder, params);
-        return builder;
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params p) {
+        XContentContext context = XContentContext.valueOf(p.param(CONTEXT_MODE_PARAM, CONTEXT_MODE_API));
+        final Iterator<? extends ToXContent> start = context == XContentContext.API
+            ? ChunkedToXContentHelper.startObject("metadata")
+            : Iterators.single((builder, params) -> builder.startObject("meta-data").field("version", version()));
+
+        final Iterator<? extends ToXContent> persSettings = context != XContentContext.API && persistentSettings().isEmpty() == false
+            ? Iterators.single(
+                (builder, params) -> persistentSettings().toXContent(
+                    builder.startObject("settings"),
+                    new ToXContent.MapParams(Collections.singletonMap("flat_settings", "true"))
+                ).endObject()
```

Comment:
Hmm I'd rather introduce a block and make 3 calls to the builder, this is a bit convoluted to read.

Reply:
++ fair point :)